ycmd/server_state.py

```diff
@@ -101,10 +101,12 @@ def ShouldUseGeneralCompleter( self, request_data ):
   def ShouldUseFiletypeCompleter( self, request_data ):
```

Comment:
Consider adding a comment here noting that a (bool, bool) is returned and what each of them is.